TombEngine/Objects/TR5/Entity/tr5_imp.cpp

```diff
@@ -109,29 +109,30 @@ namespace TEN::Entities::Creatures::TR5
 		orient.x += short(GetRandomControl() % (distance / 2) - (distance / 4));
 		orient.y += short(GetRandomControl() % (distance / 4) - (distance / 8));
 
-		int fxNumber = CreateNewEffect(item->RoomNumber);
+		int fxNumber = CreateNewEffect(item->RoomNumber, ID_IMP_ROCK, Pose(pos1));
 		if (fxNumber == NO_ITEM)
 			return;
 
-		auto& fx = EffectList[fxNumber];
+		auto& fx = g_Level.Items[fxNumber];
+		auto& fxInfo = GetFXInfo(fx);
 
-		fx.objectNumber = ID_IMP_ROCK;
-		fx.frameNumber = Objects[ID_IMP_ROCK].meshIndex;
+		fx.ObjectNumber = ID_IMP_ROCK;
+		fx.Animation.FrameNumber = Objects[ID_IMP_ROCK].meshIndex;
```

Comment:
I think what's generally wrong here is that mesh index is stored in `Animation.FrameNumber`. We now have explicitly stated mesh indices in `Model.MeshIndex`, this should be used instead.